### ISSUE TITLE
Problem with installing chatterbot with pip

### ISSUE THREAD

**Issue by TheCarBun** [gunthercox/ChatterBot#2295]
The error that I got:

Installing build dependencies ... error
  error: subprocess-exited-with-error

  × pip subprocess to install build dependencies did not run successfully.
  │ exit code: 1
  ╰─> [91 lines of output]
      Collecting setuptools
        Downloading setuptools-67.4.0-py3-none-any.whl (1.1 MB)
           ---------------------------------------- 1.1/1.1 MB 2.5 MB/s eta 0:00:00
      Collecting wheel<0.33.0,>0.32.0
        Downloading wheel-0.32.3-py2.py3-none-any.whl (21 kB)
      Collecting Cython
        Downloading Cython-0.29.33-py2.py3-none-any.whl (987 kB)
           -------------------------------------- 987.3/987.3 kB 6.9 MB/s eta 0:00:00
      Collecting cymem<2.1.0,>=2.0.2
        Downloading cymem-2.0.7-cp311-cp311-win_amd64.whl (28 kB)
      Collecting preshed<2.1.0,>=2.0.1
        Downloading preshed-2.0.1.tar.gz (113 kB)
           ---------------------------------------- 113.7/113.7 kB ? eta 0:00:00
        Preparing metadata (setup.py): started
        Preparing metadata (setup.py): finished with status 'done'
      Collecting murmurhash<1.1.0,>=0.28.0
        Downloading murmurhash-1.0.9-cp311-cp311-win_amd64.whl (18 kB)
      Collecting thinc<7.1.0,>=7.0.8
        Downloading thinc-7.0.8.tar.gz (1.9 MB)
           ---------------------------------------- 1.9/1.9 MB 7.2 MB/s eta 0:00:00
        Preparing metadata (setup.py): started
        Preparing metadata (setup.py): finished with status 'done'
      Collecting blis<0.3.0,>=0.2.1
        Downloading blis-0.2.4.tar.gz (1.5 MB)
           ---------------------------------------- 1.5/1.5 MB 7.1 MB/s eta 0:00:00
        Preparing metadata (setup.py): started
        Preparing metadata (setup.py): finished with status 'done'
      Collecting wasabi<1.1.0,>=0.0.9
        Downloading wasabi-0.10.1-py3-none-any.whl (26 kB)
      Collecting srsly<1.1.0,>=0.0.6
        Downloading srsly-1.0.6-cp311-cp311-win_amd64.whl (194 kB)
           ------------------------------------- 195.0/195.0 kB 11.5 MB/s eta 0:00:00
      Collecting numpy>=1.7.0
        Using cached numpy-1.24.2-cp311-cp311-win_amd64.whl (14.8 MB)
      Collecting plac<1.0.0,>=0.9.6
        Downloading plac-0.9.6-py2.py3-none-any.whl (20 kB)
      Collecting tqdm<5.0.0,>=4.10.0
        Downloading tqdm-4.64.1-py2.py3-none-any.whl (78 kB)
           ---------------------------------------- 78.5/78.5 kB 4.3 MB/s eta 0:00:00
      Collecting colorama
        Downloading colorama-0.4.6-py2.py3-none-any.whl (25 kB)
      Installing collected packages: wasabi, plac, cymem, wheel, srsly, setuptools, preshed, numpy, murmurhash, Cython, colorama, tqdm, blis, thinc   
        DEPRECATION: preshed is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
        Running setup.py install for preshed: started
        Running setup.py install for preshed: finished with status 'error'
        error: subprocess-exited-with-error

        Running setup.py install for preshed did not run successfully.
        exit code: 1

        [26 lines of output]
        C:\Users\subho\AppData\Local\Programs\Python\Python311\Lib\site-packages\setuptools\installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
          warnings.warn(
        WARNING: The wheel package is not available.
        running install
        C:\Users\subho\AppData\Local\Programs\Python\Python311\Lib\site-packages\setuptools\command\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
          warnings.warn(
        running build
        running build_py
        creating build
        creating build\lib.win-amd64-cpython-311
        creating build\lib.win-amd64-cpython-311\preshed
        copying preshed\about.py -> build\lib.win-amd64-cpython-311\preshed
        copying preshed\__init__.py -> build\lib.win-amd64-cpython-311\preshed
        creating build\lib.win-amd64-cpython-311\preshed\tests
        copying preshed\tests\test_counter.py -> build\lib.win-amd64-cpython-311\preshed\tests
        copying preshed\tests\test_hashing.py -> build\lib.win-amd64-cpython-311\preshed\tests
        copying preshed\tests\test_pop.py -> build\lib.win-amd64-cpython-311\preshed\tests
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

**Comment by bizbazboz** [comment]
You need an older version of python - 3.7.8 is the latest functioning version

**Comment by yevhen-dyachenko-papernest** [comment]
How can we contribute to add Python 3.10+?

**Comment by pete312** [comment]
As noted in the link in the deprecation warning  you can add --use-pep517 to the install or prevent pip from reaching that version. the issue seems to be preshed needs to migrate to the new packaging standard https://packaging.python.org/en/latest/tutorials/packaging-projects/

**Comment by bizbazboz** [comment]
> How can we contribute to add Python 3.10+?

I don't think we can, as the problem is with one of the dependencies in the program, which hasn't been updated for a while-i can't remember what it is called off the top of my head.

**Comment by chenwenfeng** [comment]
I use python 3.6 installed ok
